website/client/src/pages/settings/settingRows/userNameSetting.vue

```diff
@@ -0,0 +1,170 @@
+<template>
+  <fragment>
+    <tr
+      v-if="!modalVisible"
+    >
+      <td class="settings-label">
+        {{ $t("username") }}
+      </td>
+      <td class="settings-value">
+        {{ user?.auth?.local?.username }}
+      </td>
+      <td class="settings-button">
+        <a
+          class="edit-link"
+          @click.prevent="openModal()"
+        >
+          {{ $t(user?.auth?.local?.username ? 'edit' : 'add') }}
+        </a>
+      </td>
+    </tr>
+    <tr
+      v-if="modalVisible"
+      class="expanded"
+    >
+      <td colspan="3">
+        <div
+          v-once
+          class="dialog-title"
+        >
+          {{ $t("username") }}
+        </div>
+        <div
+          v-once
+          class="dialog-disclaimer"
+        >
+          {{ $t("changeUsernameDisclaimer") }}
+        </div>
+
+        <div class="input-area">
+          <validated-text-input
+            v-model="inputValue"
+            settings-label="username"
+            :is-valid="usernameValid"
+            :invalid-issues="usernameIssues"
+            @update:value="valuesChanged()"
+            @blur="restoreEmptyUsername()"
+          />
+
+          <save-cancel-buttons
+            :disable-save="usernameCannotSubmit"
+            @saveClicked="changeUser('username', cleanedInputValue)"
+            @cancelClicked="requestCloseModal()"
+          />
+        </div>
+      </td>
+    </tr>
+  </fragment>
+</template>
+
+<style lang="scss" scoped>
+@import '~@/assets/scss/colors.scss';
+
+</style>
+
+<script>
+import axios from 'axios';
+import debounce from 'lodash/debounce';
+import { mapState } from '@/libs/store';
+
+import { InlineSettingMixin } from '../components/inlineSettingMixin';
+import SaveCancelButtons from '../components/saveCancelButtons.vue';
+import ValidatedTextInput from '@/components/ui/validatedTextInput.vue';
+import { NotificationMixins } from '@/mixins/notifications';
+
+// TODO extract usernameIssues/checks to a mixin to share between this and the authForm
```

Comment:
later todo or now todo?

Reply:
thats a later one - easier to do in isolation pr instead of this huge change 😬